File: src/main/java/io/github/falon/scryingstatues/init/ModEntities.java
package io.github.falon.scryingstatues.init;

import io.github.falon.scryingstatues.ScryingStatues;
import io.github.falon.scryingstatues.entity.statue.StatueEntity;
import net.fabricmc.fabric.api.object.builder.v1.entity.FabricDefaultAttributeRegistry;
import net.minecraft.entity.EntityDimensions;
import net.minecraft.entity.EntityType;
import net.minecraft.entity.SpawnGroup;
import net.minecraft.registry.Registries;
import net.minecraft.registry.Registry;
import net.minecraft.util.Identifier;
import org.quiltmc.qsl.entity.api.QuiltEntityTypeBuilder;

public interface ModEntities {

	EntityType<StatueEntity> STATUE = Registry.register(Registries.ENTITY_TYPE, new Identifier(ScryingStatues.MOD_ID, "statue"),
		QuiltEntityTypeBuilder.create(SpawnGroup.MISC, StatueEntity::new).setDimensions(EntityDimensions.fixed(0.5F, 1.975F)).build());

	public static void initialize() {
		ScryingStatues.LOGGER.info("Initializing entities");
	}
}

File: src/main/java/io/github/falon/scryingstatues/init/ModEntities.kt
package io.github.falon.scryingstatues.init

import io.github.falon.scryingstatues.ScryingStatues
import io.github.falon.scryingstatues.entity.statue.StatueEntity
import net.minecraft.entity.EntityDimensions
import net.minecraft.entity.EntityType
import net.minecraft.entity.SpawnGroup
import net.minecraft.registry.Registries
import net.minecraft.registry.Registry
import net.minecraft.util.Identifier
import net.minecraft.world.World
import org.quiltmc.qsl.entity.api.QuiltEntityTypeBuilder

object ModEntities {
    val STATUE: EntityType<StatueEntity?> =
        Registry.register(
            Registries.ENTITY_TYPE,
            Identifier(ScryingStatues.MOD_ID, "statue"),
            QuiltEntityTypeBuilder.create(
                SpawnGroup.MISC,
            ) { variant: EntityType<StatueEntity?>?, world: World? ->
                StatueEntity(
                    variant,
                    world,
                )
            }.setDimensions(EntityDimensions.fixed(0.5f, 1.975f)).build(),
        )
}
